Zend/zend_API.c

```diff
@@ -1846,6 +1846,24 @@ ZEND_API zend_result object_init_ex(zval *arg, zend_class_entry *class_type) /*
 }
 /* }}} */
 
+ZEND_API zend_result object_init_with_constructor(zval *arg, zend_class_entry *class_type, uint32_t param_count, zval *params) /* {{{ */
```

Comment:
`class_type` can be derived from `arg`, see #14254.

Reply:
Oh, disregard that. Brainfart: You need the class_type to actually construct the object in the first place.